### PR TITLE
Update Developer Guide

### PR DESCRIPTION
**Pull request by DonnyS57**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -314,9 +314,22 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
 
-(For all use cases below, the **System** is the `AddressBook` and the **Actor** is the `user`, unless specified otherwise)
+(For all use cases below, the **System** is the `DonnaFin` application and the **Actor** is the `user`, unless specified otherwise)
 
-**Use case: Delete a person**
+**UC01: Adding a contact to DonnaFin**
+
+**MSS**
+
+1.  User types command to add contact along with the relevant details
```
Comment (Bluntsord) [inline docs/DeveloperGuide.md:323]: User chooses to add contact with relevant details. 

Rmb: We wanna leave out how its done and only mention what is done

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -314,9 +314,22 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+2.  DonnaFin announces that the contact has been successfully added.
+    Use Case ends.
+
+**Extensions**
+* 1a. The user types the command using the wrong format.
+  * 1a1. DonnaFin shows an error message.\
+         Use Case resumes at step 1.
```
Comment (Bluntsord) [inline docs/DeveloperGuide.md:330]: resumes from

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -314,9 +314,22 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+1.  User types command to add contact along with the relevant details
+2.  DonnaFin announces that the contact has been successfully added.
+    Use Case ends.
+
+**Extensions**
+* 1a. The user types the command using the wrong format.
+  * 1a1. DonnaFin shows an error message.\
+         Use Case resumes at step 1. 
+
+**UC02: Deleting a contact from DonnaFin**
```
Comment (Bluntsord) [inline docs/DeveloperGuide.md:332]: U gonna update this?  UC02 like never change other than addressbook -> donnafin

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -335,19 +348,51 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
 * 3a. The given index is invalid.
 
-    * 3a1. AddressBook shows an error message.
+    * 3a1. DonnaFin shows an error message.
 
       Use case resumes at step 2.
 
-*{More to be added}*
+**UC03: Locating a contact by name**
```
Comment (Bluntsord) [inline docs/DeveloperGuide.md:355]: Can change to filter or find. Feels abit weird ngl. But personal preference i guess. Ur call

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -335,19 +348,51 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+2. DonnaFin announces that the contact has been successfully updated and displays the new details.
+
+**Extensions**
+* 1a. The user types the command using the wrong format.
+    * 1a1. DonnaFin shows an error message.\
+      Use Case resumes at step 1.
+* 1b. The given index is invalid.
+    * 1b1. DonnaFin shows an error message.\
+      Use Case resumes at step 1.
+
```
Comment (Bluntsord) [inline docs/DeveloperGuide.md:382]: Can include US05: Help and US:06 List. Can refer to the addressbook userguide for the commands we can do and think from the commands there. List should be like wanna see everyone or smth. Then help is user wants to know what command to use or smth. Also can consider the saving part.

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -335,19 +348,51 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
 3.  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
-
-*{More to be added}*
+4.  A novice with no coding background should be able to use the address book. 
+5.  The system should respond within 100 milliseconds.
+6.  The size of the application should be no more than 100 MB.
+7.  All data should be stored locally.
+8.  The application should not need a remote server to function.
+9.  The application should not require any installer to start functioning.
+10. The GUI should appear fine for screen resolutions 1920x1080 and higher.
```
Comment (Bluntsord) [inline docs/DeveloperGuide.md:396]: main_stream OS can define? that should be it

**Review by Bluntsord — CHANGES_REQUESTED** [review]
Nice work. Left a few comments here and there.

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (Bluntsord) [reply docs/DeveloperGuide.md:323]: Please change this everywhere

File: docs/DeveloperGuide.md (hunk 6 above)
Comment (DonnyS57) [reply docs/DeveloperGuide.md:396]: It's already defined but the view is cutoff in GitHub!

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (DonnyS57) [reply docs/DeveloperGuide.md:323]: Okay

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (DonnyS57) [reply docs/DeveloperGuide.md:330]: fixed!

File: docs/DeveloperGuide.md (hunk 4 above)
Comment (DonnyS57) [reply docs/DeveloperGuide.md:355]: okay changed!

**Pull request merged**